### DISCUSSION TITLE
中国移动企业宽带 测速好像不太准（太快了）

### DISCUSSION THREAD

**Discussion by juicyroastgoose** [XIU2/CloudflareSpeedTest#318] [讨论]
网络状况：中国移动企业宽带 100M

测速好像不太准，显示出来的都太快了。ping下限设置成100后，根本找不到比这还慢的IP（这怎么可能。。。）

```
$ ./CloudflareST -tll 100
# XIU2/CloudflareSpeedTest v2.1.0

开始延迟测速（模式：TCP，端口：443，平均延迟上限：9999 ms，平均延迟下限：100 ms)
4931 / 4931 [------------------------------------------------------------------------------------------] 100.00%

[信息] 延迟测速结果 IP 数量为 0，跳过下载测速。

[信息] 完整测速结果 IP 数量为 0，跳过输出结果。

然后设置成50，可以找到

$ ./CloudflareST -tll 50
# XIU2/CloudflareSpeedTest v2.1.0

开始延迟测速（模式：TCP，端口：443，平均延迟上限：9999 ms，平均延迟下限：50 ms)
4931 / 4931 [------------------------------------------------------------------------------------------] 100.00%
开始下载测速（下载速度下限：0.00 MB/s，下载测速数量：10，下载测速队列：10）：
10 / 10 [------------------------------------------------------------------------------------------] 100.00%
IP 地址           已发送  已接收  丢包率  平均延迟  下载速度 (MB/s)
162.158.213.228   4       4       0.00    50.13     11.07
172.64.191.110    4       4       0.00    50.17     10.95
104.18.112.37     4       4       0.00    50.09     10.45
104.16.156.195    4       4       0.00    51.39     8.69
162.158.218.243   4       4       0.00    51.18     6.16
172.66.132.181    4       4       0.00    50.17     4.72
104.21.141.177    4       4       0.00    51.02     4.54
172.66.33.130     4       4       0.00    50.09     2.43
108.162.228.134   4       4       0.00    50.79     1.54
172.65.117.21     4       4       0.00    50.37     0.00

```

但是我好奇的是，很多IP，像第一个162.158.213.228，我本地无论是ping还是tcpping还是traceroute，都是不通的。ping.pe检测这个ip全球都不通（而不是被墙），但是测速脚本确能提供延迟和下载速度，这就让我一头雾水了。

**Comment by XIU2** [comment]
只有在 "中国移动企业宽带" 下才会遇到这种情况吗？
你测速后，在 "中国移动企业宽带" 网络下，手动 TCPing 不通？

大概率还是 企业网络 的问题，毕竟从来没有人反馈过该问题。也有类似的，但那些人都是 CloudflareST 测速走代理了。

如果这是一种普遍存在的问题的话，那么就不可能只有你一个人遇到。不过使用我软件的基本上都是 家庭宽带，没几个企业宽带的。我倒是遇到过一个联通企业宽带的反馈，不过那个人反馈的是：同样是联通，企业宽带链接 Cloudflare 吊打自己家。。。

总之，你提供的信息，我也是一头雾水。
我无法理解为什么会出现这种情况，超出我的认知了。。。

可能需要更多测试来排查问题原因，如果该问题只存在于该 "中国移动企业宽带" 网络下，那么可能和该企业网络有关系。

**Comment by XIU2** [comment]
我测试了这 10 个 IP：
```
162.158.213.228
172.64.191.110
104.18.112.37
104.16.156.195
162.158.218.243
172.66.132.181
104.21.141.177
172.66.33.130
108.162.228.134
172.65.117.21
```
只有 3 个是通的，无论是 CloudflareST 还是我手动 TCPing 都是这三个。
无法复现你遇到的情况。

```
# XIU2/CloudflareSpeedTest v2.1.0

开始延迟测速（模式：TCP，端口：443，平均延迟上限：9999 ms，平均延迟下限：0 ms)
10 / 10 [-------------------------------------------------------------------------------------------] 100.00%
IP 地址           已发送  已接收  丢包率  平均延迟  下载速度 (MB/s)
104.18.112.37     5       5       0.00    168.58    0.00
104.16.156.195    5       5       0.00    176.42    0.00
172.65.117.21     5       5       0.00    193.32    0.00

完整测速结果已写入 ZPing_result.txt 文件，可使用记事本/表格软件查看。
按下 回车键 或 Ctrl+C 退出。

# XIU2/CloudflareSpeedTest v2.1.0

开始延迟测速（模式：TCP，端口：443，平均延迟上限：9999 ms，平均延迟下限：0 ms)
10 / 10 [-------------------------------------------------------------------------------------------] 100.00%
IP 地址           已发送  已接收  丢包率  平均延迟  下载速度 (MB/s)
104.18.112.37     5       5       0.00    179.75    0.00
104.16.156.195    5       5       0.00    194.29    0.00
172.65.117.21     5       5       0.00    202.09    0.00

完整测速结果已写入 ZPing_result.txt 文件，可使用记事本/表格软件查看。
按下 回车键 或 Ctrl+C 退出。
```

**Comment by juicyroastgoose** [comment]
> 只有在 "中国移动企业宽带" 下才会遇到这种情况吗？ 你测速后，在 "中国移动企业宽带" 网络下，手动 TCPing 不通？
> 
> 大概率还是 企业网络 的问题，毕竟从来没有人反馈过该问题。也有类似的，但那些人都是 CloudflareST 测速走代理了。
> 
> 如果这是一种普遍存在的问题的话，那么就不可能只有你一个人遇到。不过使用我软件的基本上都是 家庭宽带，没几个企业宽带的。我倒是遇到过一个联通企业宽带的反馈，不过那个人反馈的是：同样是联通，企业宽带链接 Cloudflare 吊打自己家。。。
> 
> 总之，你提供的信息，我也是一头雾水。 我无法理解为什么会出现这种情况，超出我的认知了。。。
> 
> 可能需要更多测试来排查问题原因，如果该问题只存在于该 "中国移动企业宽带" 网络下，那么可能和该企业网络有关系。

我家里的电信家庭宽带，是正常的。我看了一下IP地址，移动确实有很多CF的节点在香港，所以ping很快，只有几十ms。但是不太应该“所有”IP都很快，这点比较奇怪。。。

另外，那些ping不通的ip，不是光国内ping/tcpping不通，我从国外的vps上ping这些ip，也是不通的，就好像不存在这些节点一样，但是不清楚脚本为什么会把这些ip选择出来。。。

总之，移动的网络很奇怪。。。

**Comment by XIU2** [comment]
如果该问题只存在于该 "中国移动企业宽带" 网络下，那么多半和该企业网络有什么关系了~

或许是有劫持？
我没用过企业宽带，不清楚企业宽带有什么条条道道。

另外，你没有回答：`你测速后，在 "中国移动企业宽带" 网络下，手动 TCPing 不通？`
即，在 "中国移动企业宽带" 网络下，你手动 TCPing 和 CloudflareST 测速结果是否一致？

**Comment by juicyroastgoose** [comment]
> 如果该问题只存在于该 "中国移动企业宽带" 网络下，那么多半和该企业网络有什么关系了~
> 
> 或许是有劫持？ 我没用过企业宽带，不清楚企业宽带有什么条条道道。
> 
> 另外，你没有回答：`你测速后，在 "中国移动企业宽带" 网络下，手动 TCPing 不通？` 即，在 "中国移动企业宽带" 网络下，你手动 TCPing 和 CloudflareST 测速结果是否一致？

能ping通的IP，数值和CloudflareST测试出来是一致的（ping和tcpping都试过）。但是不能ping通的ip，ping和tcpping都是timeout，只有CloudflareST可以测试出来结果。

应该不是劫持，我用besttrace看了一下到104.18.112.37路由，移动确实是到了cloudflare香港节点。但同一个IP，电信traceroute到的是美国。。。

**Comment by XIU2** [comment]
那就奇怪了，CloudflareST 是直接建立 TCP 链接来 TCPing 的，和其他 TCPing 工具没啥区别，怎么会不一样呢。。。

你把这些 IP 单独放到一个 1.txt 文件内，CloudflareST 测速时指定 `-f 1.txt` 来对其测速看看。

**Comment by juicyroastgoose** [comment]
> 那就奇怪了，CloudflareST 是直接建立 TCP 链接来 TCPing 的，和其他 TCPing 工具没啥区别，怎么会不一样呢。。。
> 
> 你把这些 IP 单独放到一个 1.txt 文件内，CloudflareST 测速时指定 `-f 1.txt` 来对其测速看看。

我把这10个IP放在单独的文件测试，但是这样数字就更夸张了。。。

$ ./CloudflareST -f ip.txt
# XIU2/CloudflareSpeedTest v2.1.0

开始延迟测速（模式：TCP，端口：443，平均延迟上限：9999 ms，平均延迟下限：0 ms)
10 / 10 [------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------] 100.00%
开始下载测速（下载速度下限：0.00 MB/s，下载测速数量：10，下载测速队列：10）：
10 / 10 [------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------] 100.00%
IP 地址           已发送  已接收  丢包率  平均延迟  下载速度 (MB/s)
172.66.33.130     4       4       0.00    0.31      10.66
108.162.228.134   4       4       0.00    0.32      10.37
162.158.218.243   4       4       0.00    1.97      10.29
104.21.141.177    4       4       0.00    1.12      9.58
104.16.156.195    4       4       0.00    0.70      9.36
172.66.132.181    4       4       0.00    1.18      8.48
104.18.112.37     4       4       0.00    0.53      8.10
172.65.117.21     4       4       0.00    1.66      4.90
162.158.213.228   4       4       0.00    0.88      4.25
172.64.191.110    4       4       0.00    0.73      0.00

**Comment by c15412** [comment]
都这样了应该是劫持吧，现在运营商的劫持技术超乎你的想象

**Comment by c15412** [comment]
感觉是程序被移动精准识别然后主动劫持了

**Comment by XIU2** [comment]
这个延迟，肯定是假的。
要么走代理了，要么被劫持了。

**Comment by juicyroastgoose** [comment]
可能是吧。。。企业宽带为了速度更快，移动自建内网cdn给CF？也就是变成了 用户->移动内网CDN（劫持所有CF的IP？）->CF的CDN节点->目标服务器？

**Comment by XIU2** [comment]
如果是走代理了，那么对国外来说，这个延迟就是正常的，都是 Anycast 节点。
国外服务器去测速看到的 IP，都是这样延迟很低的，因为都有就近节点。

**Comment by c15412** [comment]
> 可能是吧。。。企业宽带为了速度更快，移动自建内网cdn给CF？也就是变成了 用户->移动内网CDN（劫持所有CF的IP？）->CF的CDN节点->目标服务器？

应该是的，就和移动之前给bt下载搭建内网缓存一样

**Comment by MOOnLyer** [comment]
联通家宽

今天一试也是70左右的ping 自己试了下也是能ping通的

看了下traceroute 走的是香港CDN 目前使用正常
